src/main/java/com/iota/iri/cache/Cache.java

```diff
@@ -102,9 +103,9 @@
     CacheConfiguration getConfiguration();
 
     /**
-     * Poll and return the next key to release from cache
+     * Gets the release queue
      * 
-     * @return The key to release
+     * @return The release queue
      */
-    K nextReleaseKey();
+    Queue<K> getReleaseQueue();
```

Comment:
Why did you do this change, you're disobeying the law of demeter... (see [styleguide](https://github.com/iotaledger/iri/blob/dev/STYLEGUIDE.md#obey-the-law-of-demeter-lod))

Reply:
It's the only way if we want to use saveBatch

Reply:
As of now, we will only make use of it during shutdown, which i dont think is worth it. Currently savebatch is only unsed when rescanning and metadata saving which only happens once. I think its best to revert this getReleaseQueue (Sorry bill! :D)

Reply:
😅

Reply:
Lets see what gal thinks about it!

Reply:
When we solidify several txs at once it is used... you can argue that it is not worth it
It is a preparation for the future where we will also do save batch on new txs.
In the beginning it was my original intention with the change, but then I understood that it is too risky for the releases...

Anyhows, I think we should stick with it. But I think Bill should still obey the LoD.
Why not just have a `pollReleaseKey` method?

Reply:
We don't want to `poll` from release queue if we might not release it.

Reply:
Simply follow LoD by writing methods that delegate the queue please

Reply:
no class that holds the `cache` needs to know about its internal members